src/launch/service.c

```diff
@@ -399,19 +399,30 @@ static int service_watch_unit_handler(sd_bus_message *message, void *userdata, s
 }
 
 static int service_watch_unit(Service *service, const char *unit) {
-        _c_cleanup_(c_freep) char *object_path = NULL;
+        const char *object_path = NULL;
+        _c_cleanup_(sd_bus_message_unrefp) sd_bus_message *m = NULL;
         int r;
 
         assert(!service->slot_watch_unit);
 
-        r = sd_bus_path_encode(
-                "/org/freedesktop/systemd1/unit",
-                unit,
-                &object_path
+        r = sd_bus_call_method(
```

Comment:
other calls into pid1 here are async - are we sure it's ok to block on this one?

Reply:
`org.freedesktop.systemd1.Manager.LoadUnit` should at most cause a single filesystem access in pid1, so it should complete within a reasonably short amount of time.

Reply:
You cannot do blocking calls into systemd here. This is not about the costs, but rather about circular dependencies. You easily run into deadlocks, like if systemd tries to call into polkit before answering the request. There is so many possibilities how this can go wrong and dbus-broker ends up waiting on systemd, which ends up waiting on D-Bus to serve the request.

Preferably, I would not to that call to LoadUnit(), unless really necessary. It definitely is not trivial and I would prefer a separate PR and discussion, because the other fix looks fine and could be merged.

Reply:
Would just converting the call to asynchronous be sufficient?

Reply:
Seems like it would be possible to get the unit object path from the `org.freedesktop.systemd1.Job` object instead.